### ISSUE TITLE
Using compiler-cli (ngc) with Webpack for AOT Compilation

### ISSUE THREAD

**Issue by sime** [angular/angular.io#2099]
Since RC5 Angular recommends AOT Compilation for production builds, the current Webpack guide doesn't mention or include compiler-cli or ngc in `webpack.prod.js`.

I'm imagining that this is being blocked by a missing `ngc` loader.

**Comment by bogomips** [comment]
+1

**Comment by mkoestner** [comment]
+1

**Comment by Foxandxss** [comment]
That is not possible right now. A few things needs to be changed so we can use AoT with webpack like that.

Check https://github.com/angular/angular/issues/8759 for more info.

I am going to close this, but be sure that I would update the guide when possible.

**Comment by TheLarkInn** [comment]
Despite being reported 3 months ago, this is still not possible due to a lack of API support on both AoT compiler and the current Typescript loaders. 

Please read the following issue _and related tied issues_ to get a full picture of the state of integration:
https://github.com/angular/angular/issues/8759

**Comment by bogomips** [comment]
what about systemJs? is it possible with it?

**Comment by TheLarkInn** [comment]
I'm sorry let me rephrase, does it integrate with webpack the _right way_? No.

Can you just run it as a previous build step and let webpack bundle the transpiled js and it will work? Yes.

**Comment by Martin-Wegner** [comment]
@TheLarkInn  can you extend the webpack tutorial with a how to 'you just run it as a previous build step and let webpack bundle the transpiled js' please?

**Comment by TheLarkInn** [comment]
Well simply put: AoT transpiles your app into super efficient evaluative code, you simply webpack the same code but pointing towards the js file entry point.

**Comment by freshprinze** [comment]
@Martin-Wegner @TheLarkInn Check https://github.com/mgechev/angular2-seed/tree/offline-compilation. This uses SystemJS as the bundler; but @mgechev used the AoT compiler in compile.ahead.prod.ts. Maybe this will shed some light on how this can be achieved as a separate build step.

**Comment by IAMtheIAM** [comment]
Where is the file `compile.ahead.prod.ts` located? I couldnt find it anywhere to see what you are referring to.

**Comment by TheLarkInn** [comment]
All it is, is doing it in a separate build step. That is not the direction we want to go or _should_ go for webpack users. We are actively working on adding this feature.

**Comment by freshprinze** [comment]
@IAMtheIAM Check https://github.com/mgechev/angular2-seed/blob/e8105f1627a752b2bae2d1d575a65beecf982c8c/tools/tasks/seed/compile.ahead.prod.ts. 

As @TheLarkInn suggested there should be a _right way_ to do this straight from webpack.

**Comment by PraveenPeri** [comment]
Any update on this?

**Comment by neha-s16** [comment]
@TheLarkInn Is this feature there yet?

**Comment by Foxandxss** [comment]
The guide will be updated with AOT support in early February.

**Comment by neha-s16** [comment]
Great Thanks!

**Comment by JaredDanilson** [comment]
@Foxandxss Do you have a revised ETA when the guide will be updated with AOT support?  Thanks!
